source/change-streams/change-streams.rst

```diff
@@ -43,35 +43,60 @@ Resumable Error
 
 An error is considered resumable if it meets any of the following criteria:
 
-- any error encountered which is not a server error (e.g. a timeout error or
+- Any error encountered which is not a server error (e.g. a timeout error or
   network error)
 
-- *any* server error response from a getMore command excluding those
-  containing the error label `NonResumableChangeStreamError` and those
-  containing the following error codes
+- For servers with wire version 9 or higher (server version 4.4 or higher), any
+  server error with the `ResumableChangeStreamError` error label.
+
+- For servers with wire version less than 9, a server error with one of the
+  following codes:
 
   .. list-table::
     :header-rows: 1
 
     * - Error Name
       - Error Code
-    * - Interrupted
-      - 11601
-    * - CappedPositionLost
-      - 136
-    * - CursorKilled
-      - 237
+    * - HostUnreachable
+      - 6
+    * - HostNotFound
+      - 7
+    * - NetworkTimeout
+      - 89
+    * - ShutdownInProgress
+      - 91
+    * - PrimarySteppedDown
+      - 189
+    * - ExceededTimeLimit
+      - 262
+    * - SocketException
+      - 9001
+    * - NotMaster
+      - 10107
+    * - InterruptedAtShutdown
+      - 11600
+    * - InterruptedDueToReplStateChange
+      - 11602
+    * - NotMasterNoSlaveOk
+      - 13435
+    * - NotMasterOrSecondary
+      - 13436
+    * - StaleShardVersion
+      - 63
+    * - StaleEpoch
+      - 150
+    * - StaleConfig
+      - 13388
+    * - RetryChangeStream
+      - 234
+    * - FailedToSatisfyReadPreference
+      - 133
+    * - ElectionInProgress
+      - 216
 
 An error on an aggregate command is not a resumable error. Only errors on a
 getMore command may be considered resumable errors.
 
-The criteria for resumable errors is similar to the discussion in the SDAM
```

Comment:
This is here because the spec used to explicitly call out that error messages containing the substring "not master" or "node is recovering" as resumable. (libmongoc currently the check as well).

I looked through the history:
https://jira.mongodb.org/browse/SPEC-1051 introduced the "not master" and "node is recovering" criteria:
> - a server error response without an error code or one different from those
  listed above, but with an error message containing the substring "not
  master" or "node is recovering"

https://jira.mongodb.org/browse/SPEC-1124 removed the bullet

With this new change I think you can close https://jira.mongodb.org/browse/SPEC-1328 ("Change stream spec need not discuss "not master" errors") as "Gone Away"

If it seems out of scope we can defer, but I'd vote for adding another test that checks an error that is not on the whitelist but has the error message "not master" is not resumed.

Reply:
I read through the spec tickets you linked and some of our specs do still mention checking for the error message substrings. The recent work for using error labels in retryable writes removed that check from the retryable writes work because nobody on the drivers or server team thought it was necessary to have checks for error codes not existing.